File: 99_Bottles_Problem/Javascript/mariolamacchia/99-bottles.js
function bottlesOfBeer(i) {
  var s;
  if (i == 1) {
    s = "1 bottle";
  } else if (i == 0) {
    s = "no more bottles";
  } else {
    s = i + " bottles";
  }
  return s + ' of beer';
}

var maxBottles = 99;
var string = '';
for (var i = maxBottles; i > 0; i--) {
  string += bottlesOfBeer(i) + ' on the wall, ' + bottlesOfBeer(i) + '.\n';
  string += 'Take one down, pass it around, ' + bottlesOfBeer(i-1) + ' on the wall.\n';
}
string += 'No more bottles of beer on the wall, no more bottles of beer.\n';
string += 'Go to the store and buy some more, ' + bottlesOfBeer(maxBottles) + ' on the wall.';

console.log(string);

File: 99_Bottles_Problem/Python/jiang42/99_bottles.py
#!/usr/bin/env python

def bottles():
  bottles = 99
  while bottles > 0:
    print("{0} {1} of beer on the wall, {0} {1} of beer.".format(bottles if bottles != 0 else "no more", "bottles" if bottles != 1 else "bottle"))
    print("Take one down and pass it around, {0} {1} of beer on the wall.".format("no more" if bottles == 1 else bottles - 1, "bottle" if bottles == 2 else "bottles"))
    bottles -= 1
  print("No more bottles of beer on the wall, no more bottles of beer.")
  print("Go to the store and buy some more, 99 bottles of beer on the wall.")

if __name__ == '__main__':
  bottles()
